src/oas3/service.ts

```diff
@@ -67,13 +69,28 @@ export const transformOas3Service: Oas3HttpServiceTransformer = ({ document }) =
       if (!sec) return null;
 
       return keys(sec).map(key => {
-        return securitySchemes.find(securityScheme => {
-          return securityScheme.key === key;
-        });
+        const ss = securitySchemes.find(securityScheme => securityScheme.key === key);
+        if (ss && ss.type === 'oauth2') {
+          return {
+            ...ss,
+            flows: mapValues(ss.flows, (flow: OAuthFlowObject) => ({
+              ...flow,
+              scopes: pickBy(flow.scopes, (_val: string, scopeKey: string) => {
+                const secKey = sec[key];
+                if (secKey) return secKey.includes(scopeKey);
+                return false;
+              }),
+            })),
+          };
+        }
+
+        return ss;
       });
     }),
   );
+
   if (security.length) {
+    //@ts-ignore
```

Comment:
I hate doing this, but Lodash types are effectively very loose and would require other checks that we do not really need.